04_Intro_to_Butler.ipynb

```diff
@@ -426,39 +473,58 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "ra = 70.37770\n",
-    "dec = -37.1757\n",
-    "s1 = \"exposure.day_obs = 20251201\"\n",
-    "s2 = \"exposure.tracking_ra > \"+str(ra-1.0)\n",
-    "s3 = \"exposure.tracking_ra < \"+str(ra+1.0)\n",
-    "s4 = \"exposure.tracking_dec > \"+str(dec-1.0)\n",
-    "s5 = \"exposure.tracking_dec < \"+str(dec+1.0)\n",
-    "\n",
-    "results = registry.queryDimensionRecords('visit',\n",
-    "                                         where=s1+\" AND \"+s2+\" AND \"+s3+\" AND \"+s4+\" AND \"+s5,\n",
-    "                                         collections=collection)\n",
+    "import lsst.sphgeom\n",
     "\n",
-    "# Use expandDataId to fill in the implicit dataId keys with values\n",
-    "for i, ref in enumerate(results):\n",
-    "    tempId = butler.registry.expandDataId(ref.dataId)\n",
-    "    print(tempId.full)\n",
-    "    if i > 10:\n",
+    "pixelization = lsst.sphgeom.HtmPixelization(7)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "htm_id = pixelization.index(\n",
+    "    lsst.sphgeom.UnitVector3d(\n",
+    "        lsst.sphgeom.LonLat.fromDegrees(70.37699524983329, -37.17573628348882)\n",
+    "    )\n",
+    ")\n",
+    "scale = pixelization.triangle(htm_id).getBoundingCircle().getOpeningAngle().asDegrees()*3600\n",
+    "print(f'HTM ID={htm_id} at level={pixelization.getLevel()} is a ~{scale:0.2}\" triangle.')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And we can use that to query for (e.g.) the set of all `src` data products that overlap this point in i:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for i, src_ref in enumerate(registry.queryDatasets(\"src\", htm20=htm_id, band=\"i\")):\n",
```

Comment:
We still want to retain the `exposure.day_obs = YYYYMMDD` for this example I think?

I was also going to say that you really don't want to build up where clauses like was being done in the previous version. That's what the `bind` argument is designed for.

Reply:
That might be useful as a demostration of how to query on other fields, but I think using `day_obs` is an antipattern for temporal searches; in order to use it precisely you have to know how it's defined, and it's not indexed (though it probably should be).  If you use timespans then astropy handles the time system definitions for you and you get good indexes.

Reply:
day_obs is standardized to be YYYYMMDD (the metadata translator insists on that) so I don't think that's a problem. Timespans are clearly the right answer in general but there are some cases where a quick YYYYMMDD is much more convenient (and it's the only thing you can do in a dataId with a seqnum). If we want to show records being used to constrain queries there are other options such as restricting the airmass.

Reply:
I was thinking about the definition in terms of the time of day and timezone that `day_obs` turned over, but maybe my recollection that that was somehow tricky or non-standard was wrong anyway.

Reply:
We define it to make sense for the instrument that it refers to. For LSST we shift the day obs definition by a few hours such that it doesn't change during the night. In Hawaii the night is always the same UT throughout.